### PR TITLE
Purview datamap

### PR DESCRIPTION
**Pull request by yifan-zhou922**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/purview/azure-analytics-purview-datamap/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+### Authentication
+
+[Azure Identity][azure_identity] package provides the default implementation for authenticating the client.
+
+## Key concepts
+
+## Examples
+
+```java com.azure.analytics.purview.datamap.readme
```
Comment (haolingdong-msft) [inline sdk/purview/azure-analytics-purview-datamap/README.md:41]: no code?

File: sdk/purview/azure-analytics-purview-datamap/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+## Troubleshooting
```
Comment (haolingdong-msft) [inline sdk/purview/azure-analytics-purview-datamap/README.md:44]: Add contents.

File: sdk/purview/azure-analytics-purview-datamap/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Azure PurviewDataMap client library for Java
+
+Azure PurviewDataMap client library for Java.
+
+This package contains Microsoft Azure PurviewDataMap client library.
```
Comment (haolingdong-msft) [inline sdk/purview/azure-analytics-purview-datamap/README.md:5]: Please be more descriptive on what does purview datamap do and what does the library do.

Comment (yifan-zhou922) [reply sdk/purview/azure-analytics-purview-datamap/README.md:5]: updated

File: sdk/purview/azure-analytics-purview-datamap/README.md (hunk 1 above)
Comment (yifan-zhou922) [reply sdk/purview/azure-analytics-purview-datamap/README.md:41]: updated

File: sdk/purview/azure-analytics-purview-datamap/README.md (hunk 2 above)
Comment (yifan-zhou922) [reply sdk/purview/azure-analytics-purview-datamap/README.md:53]: updated

File: sdk/purview/azure-analytics-purview-datamap/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+## Examples
+
+```java com.azure.analytics.purview.datamap.readme
+ClientSecretCredential cred = new ClientSecretCredentialBuilder()
+            .tenantId(Configuration.getGlobalConfiguration().get("TENANT_ID"))
+            .authorityHost(Configuration.getGlobalConfiguration().get("AUTHORITY_HOST"))
+            .clientId(Configuration.getGlobalConfiguration().get("CLIENT_ID"))
+            .clientSecret(Configuration.getGlobalConfiguration().get("CLIENT_SECRET"))
+            .build();
+        DataMapClientBuilder clientBuilder = new DataMapClientBuilder()
```
Comment (haolingdong-msft) [inline sdk/purview/azure-analytics-purview-datamap/README.md:48]: remove the indent?

Comment (yifan-zhou922) [reply sdk/purview/azure-analytics-purview-datamap/README.md]: Thanks. Updated

**Comment by yifan-zhou922** [comment]
Hi @haolingdong-msft , please help with the ci failures.

**Comment by yifan-zhou922** [comment]
I have updated the recorded tests according to the TSG. And the local test passed. 
image
image

File: eng/versioning/version_client.txt [hunk 5]
```diff
@@ -51,6 +51,7 @@ com.azure:azure-ai-textanalytics-perf;1.0.0-beta.1;1.0.0-beta.1
 com.azure:azure-ai-translation-text;1.0.0-beta.1;1.0.0-beta.2
 com.azure:azure-analytics-defender-easm;1.0.0-beta.1;1.0.0-beta.2
 com.azure:azure-analytics-purview-catalog;1.0.0-beta.4;1.0.0-beta.5
+com.azure:azure-analytics-purview-datamap;1.0.0-beta.1
```
Comment (mssfang) [inline eng/versioning/version_client.txt:54]:
```suggestion
com.azure:azure-analytics-purview-datamap;1.0.0-beta.1;1.0.0-beta.1
```

**Comment by yifan-zhou922** [comment]
Some other ci test failures seem to be related with other packages.

**Comment by haolingdong-msft** [comment]
Hi @yifan-zhou922, as discussed previously, you will need to fix the CI. Current there is still CI failures. https://dev.azure.com/azure-sdk/public/_build/results?buildId=3532055&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=ec247c96-ff2a-56b3-43b7-8e8a4c52422f&l=33 -> This is due to the wrong javadoc `@search`, I don't think javadoc has `@search` tag.

File: sdk/purview/azure-analytics-purview-datamap/README.md (hunk 2 above)
Comment (haolingdong-msft) [reply sdk/purview/azure-analytics-purview-datamap/README.md:53]: Still don't see the content of 'Troubleshooting'. Where did you update at?

File: sdk/purview/azure-analytics-purview-datamap/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    <version>1.0.0-beta.1</version>
+</dependency>
+```
+[//]: # ({x-version-update-end})
+
+### Authentication
+
+[Azure Identity][azure_identity] package provides the default implementation for authenticating the client.
+
+## Key concepts
```
Comment (haolingdong-msft) [inline sdk/purview/azure-analytics-purview-datamap/README.md:37]: Add contents for 'Key concepts'

Comment (yifan-zhou922) [reply sdk/purview/azure-analytics-purview-datamap/README.md:37]: updated

File: sdk/purview/azure-analytics-purview-datamap/CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Release History
+
+## 1.0.0-beta.1 (2024-03-01)
+
+- Azure PurviewDataMap client library for Java. This package contains Microsoft Azure PurviewDataMap client library.
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-datamap/CHANGELOG.md:5]: nit, Purview Data Map

**Review by weidongxu-microsoft — COMMENTED** [review]
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3552170&view=logs&j=3dffcd12-e604-5fb1-810a-84e3eaee15b4&t=d3b306d1-76ab-5535-a5f7-f8bdc21a8bfb&l=41

see the Quickstart Troubleshoot doc about set a lower coverage

Note that GA requires you have the high coverage.

---

Not a blocker, but please have more samples (and a readme.md).
e.g. https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/contentsafety/azure-ai-contentsafety/src/samples

**Comment by haolingdong-msft** [comment]
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3552170&view=logs&j=36df4387-3551-57fa-098c-b43a23930261&t=3851d55a-4253-57f5-016e-927e8de9d21b&l=13
CI error is due to the dependency version. After you merge with latest main branch, the error should be solved.

File: sdk/purview/azure-analytics-purview-datamap/tsp-location.yaml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+directory: specification/purview/Azure.Analytics.Purview.DataMap
+commit: e4dd3e7e4d0402a81b2bef7dd754d3e46e8a8ab5
+repo: Azure/azure-rest-api-specs
+cleanup: false
```
Comment (haolingdong-msft) [inline sdk/purview/azure-analytics-purview-datamap/tsp-location.yaml:4]: should we remove 'cleanup: false'

**Comment by haolingdong-msft** [comment]
```
[ERROR] Failed to execute goal com.azure.tools:codesnippet-maven-plugin:1.0.0-beta.8:verify-codesnippet (default-cli) on project azure-analytics-purview-datamap: codesnippet-maven-plugin has encountered errors while verifying codesnippets.
[ERROR]
[ERROR] The following codesnippets need updates:
[ERROR] SnippetId: com.azure.analytics.purview.datamap.readme, Location: /mnt/vss/_work/1/s/sdk/purview/azure-analytics-purview-datamap/README.md
[ERROR] -> [Help 1]
```
This CI error is due to your code is updated, but your code snippet in  readme is not updated. You can update your local codesnippet and push the changes.

**Comment by yifan-zhou922** [comment]
New failures:
`[ERROR] Failed to execute goal on project azure-analytics-purview-catalog: Could not resolve dependencies for project com.azure:azure-analytics-purview-catalog:jar:1.0.0-beta.5: Failed to collect dependencies at com.azure:azure-identity:jar:1.11.3 -> com.microsoft.azure:msal4j:jar:1.14.0 -> com.nimbusds:oauth2-oidc-sdk:jar:10.7.1 -> net.minidev:json-smart:jar:2.1.0: Failed to read artifact descriptor for net.minidev:json-smart:jar:2.1.0: net.minidev:json-smart:pom:2.1.0 failed to transfer from https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced. Original error: Could not transfer artifact net.minidev:json-smart:pom:2.1.0 from/to central (https://repo.maven.apache.org/maven2): /mnt/vss/_work/1/.m2/repository/net/minidev/json-smart/2.1.0/json-smart-2.1.0.pom.part (No such file or directory) -> [Help 1]
[ERROR] Failed to execute goal on project azure-analytics-purview-datamap: Could not resolve dependencies for project com.azure:azure-analytics-purview-datamap:jar:1.0.0-beta.1: Failed to collect dependencies at com.azure:azure-identity:jar:1.11.3 -> com.microsoft.azure:msal4j:jar:1.14.0 -> com.nimbusds:oauth2-oidc-sdk:jar:10.7.1 -> com.nimbusds:lang-tag:jar:1.7: Failed to read artifact descriptor for com.nimbusds:lang-tag:jar:1.7: com.nimbusds:lang-tag:pom:1.7 failed to transfer from https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced. Original error: Could not transfer artifact com.nimbusds:lang-tag:pom:1.7 from/to central (https://repo.maven.apache.org/maven2): /mnt/vss/_work/1/.m2/repository/com/nimbusds/lang-tag/1.7/lang-tag-1.7.pom.part (No such file or directory) -> [Help 1]
[ERROR] Failed to execute goal on project azure-analytics-purview-workflow: Could not resolve dependencies for project com.azure:azure-analytics-purview-workflow:jar:1.0.0-beta.2: Failed to collect dependencies at com.azure:azure-identity:jar:1.11.3 -> com.microsoft.azure:msal4j:jar:1.14.0 -> com.nimbusds:oauth2-oidc-sdk:jar:10.7.1 -> net.minidev:json-smart:jar:2.1.0: Failed to read artifact descriptor for net.minidev:json-smart:jar:2.1.0: net.minidev:json-smart:pom:2.1.0 failed to transfer from https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced. Original error: Could not transfer artifact net.minidev:json-smart:pom:2.1.0 from/to central (https://repo.maven.apache.org/maven2): /mnt/vss/_work/1/.m2/repository/net/minidev/json-smart/2.1.0/json-smart-2.1.0.pom.part (No such file or directory) -> [Help 1]
[ERROR] Failed to execute goal on project azure-analytics-purview-scanning: Could not resolve dependencies for project com.azure:azure-analytics-purview-scanning:jar:1.0.0-beta.3: Failed to collect dependencies at com.azure:azure-identity:jar:1.11.3 -> com.microsoft.azure:msal4j:jar:1.14.0 -> com.nimbusds:oauth2-oidc-sdk:jar:10.7.1 -> net.minidev:json-smart:jar:2.1.0: Failed to read artifact descriptor for net.minidev:json-smart:jar:2.1.0: Could not transfer artifact net.minidev:json-smart:pom:2.1.0 from/to central (https://repo.maven.apache.org/maven2): /mnt/vss/_work/1/.m2/repository/net/minidev/json-smart/2.1.0/json-smart-2.1.0.pom.part (No such file or directory) -> [Help 1]
[ERROR] `
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3553651&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=b5f103f3-7317-514c-e60c-6e19642365f6

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by haolingdong-msft** [comment]
https://github.com/Azure/azure-sdk-for-java/blob/ec12185d74f30e2db9c0501510bf86986e8bd0bb/sdk/purview/azure-analytics-purview-datamap/README.md
image

readme still not look good, could you please resolve below comments:
1. example is empty. Please add the example.
2. add contents for trouble shooting section.

**Comment by yifan-zhou922** [comment]
> https://github.com/Azure/azure-sdk-for-java/blob/ec12185d74f30e2db9c0501510bf86986e8bd0bb/sdk/purview/azure-analytics-purview-datamap/README.md image
> 
> readme still not look good, could you please resolve below comments:
> 
> 1. example is empty. Please add the example.
> 2. add contents for trouble shooting section.

Thanks. We have the example section. https://github.com/yifan-zhou922/azure-sdk-for-java/tree/yifanzhou/datamap/sdk/purview/azure-analytics-purview-datamap

I notice other purview packages do not have trouble shooting sections yet. Is this a blocker? Can we add more trouble shooting as the follow-up itme?

**Review by haolingdong-msft — APPROVED** [review]
Approved. Trouble shooting is not a blocker for preview release, but please do add later.

**Pull request merged**